litmus-portal/graphql-server/pkg/k8s/cluster.go

```diff
@@ -71,50 +74,148 @@ func ClusterResource(manifest string, namespace string) (*unstructured.Unstructu
+		ServerServiceName = os.Getenv("SERVER_SERVICE_NAME")
+		NodeName          = os.Getenv("NODE_NAME")
+		LitmusPortalNS    = os.Getenv("LITMUS_PORTAL_NAMESPACE")
+		Ingress           = os.Getenv("INGRESS")
+		IngressName       = os.Getenv("INGRESS_NAME")
 	)
 
 	clientset, err := GetGenericK8sClient()
 	if err != nil {
 		return "", err
 	}
 
-	podList, _ := clientset.CoreV1().Pods(LitmusPortalNS).List(metaV1.ListOptions{
-		LabelSelector: "component=litmusportal-server",
-	})
+	if Ingress == "true" {
+		getIng, err := clientset.ExtensionsV1beta1().Ingresses(LitmusPortalNS).Get(IngressName, metaV1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
 
-	svc, err := clientset.CoreV1().Services(LitmusPortalNS).Get("litmusportal-server-service", metaV1.GetOptions{})
-	if err != nil {
-		return "", err
-	}
+		/*
+			Priorities of retrieving Ingress endpoint
+			1. hostname
+			2. IPAddress
+		*/
+
+		if len(getIng.Status.LoadBalancer.Ingress) > 0 {
+			if len(getIng.Spec.Rules) > 0 {
+				if getIng.Spec.Rules[0].Host != "" {
+					IPAddress = getIng.Spec.Rules[0].Host
+				} else if getIng.Status.LoadBalancer.Ingress[0].IP != "" {
+					IPAddress = getIng.Status.LoadBalancer.Ingress[0].IP
+				}
+			} else {
+				return "", errors.New("Ingress rules are not present")
+			}
+		} else {
+			return "", errors.New("IP Address or HostName not generated")
+		}
 
-	for _, port := range svc.Spec.Ports {
-		if port.Name == "graphql-server" {
-			Nodeport = port.NodePort
+		if IPAddress == "" {
+			return "", errors.New("IP Address or Hostname is not available in the ingress of " + IngressName)
 		}
-	}
 
-	nodeIP, err := clientset.CoreV1().Nodes().Get(podList.Items[0].Spec.NodeName, metaV1.GetOptions{})
-	if err != nil {
-		return "", err
-	}
+		for _, rule := range getIng.Spec.Rules {
+			for _, path := range rule.HTTP.Paths {
+				if path.Backend.ServiceName == ServerServiceName {
+					f := func(c rune) bool {
+						return c == '/'
+					}
+
+					path_arr := strings.FieldsFunc(path.Path, f)
+					if len(path_arr) > 0 {
+						if path_arr[len(path_arr)-1] == "(.*)" {
+							path_arr[len(path_arr)-1] = "query"
+						} else {
+							path_arr = append(path_arr, "query")
+						}
+					} else {
+						path_arr = append(path_arr, "query")
+					}
+
+					IngressPath = strings.Join(path_arr[:], "/")
+				}
+			}
+		}
 
-	for _, addr := range nodeIP.Status.Addresses {
-		if addr.Type == "ExternalIP" && addr.Address != "" {
-			ExternalIP = addr.Address
-		} else if addr.Type == "InternalIP" && addr.Address != "" {
-			InternalIP = addr.Address
+		if len(getIng.Spec.TLS) > 0 {
+			Scheme = "https"
+		} else {
+			Scheme = "http"
 		}
-	}
 
-	if ExternalIP == "" {
-		return "http://" + InternalIP + ":" + strconv.Itoa(int(Nodeport)), nil
-	} else {
-		return "http://" + ExternalIP + ":" + strconv.Itoa(int(Nodeport)), nil
+		FinalUrl = Scheme + "://" + IPAddress + "/" + IngressPath
+
+	} else if Ingress == "false" || Ingress == "" {
```

Comment:
if Ingress env is something else apart from false or "" like a wrong input we should either throw error or default to false

Reply:
Done, added a condition
```else if Ingress != "false" || Ingress != "" || Ingress != "true" {
		return "", errors.New("Ingress value is not correct")
	}
```

Reply:
Don't see the change

Reply:
Line no-218

Reply:
I don't see it https://github.com/litmuschaos/litmus/blob/6b90f91604fa10d1e33389c0384c76346aebbc03/litmus-portal/graphql-server/pkg/k8s/cluster.go#L218

Reply:
Can you refresh your browser

Reply:
https://github.com/litmuschaos/litmus/blob/d8647bb167bd7cbbf3aa4074a97ff69840e89d6f/litmus-portal/graphql-server/pkg/k8s/cluster.go#L218

Reply:
I see some issue with github. I gues we can make it just `else` instead of another `else if`

Reply:
Done